typing/typedtree.ml

```diff
@@ -845,3 +845,19 @@ let split_pattern pat =
         combine_opts (into cpat) exns1 exns2
   in
   split_pattern pat
+
+(* Expressions are considered short if they satisfy one of:
+
+   - Looking like an identifier, small words separated with a '.' or a '#'.
+   - Do not contain spaces when printed.
+   - Look nice as the subject of a sentence, eg. "'Foo.bar' is short." *)
+let rec exp_is_short exp =
+  match exp.exp_desc with
+  | _ when exp.exp_attributes <> [] -> false
+  | Texp_ident _ | Texp_instvar _ | Texp_constant _
+  | Texp_variant (_, None)
+  | Texp_construct (_, _, [])
+  | Texp_unreachable ->
+      true
```

Comment:
I would propose to exlude the `Texp_unreachable` because without enough context, it is hard to know that `.` means `unreachable branch`.

Reply:
Done.